tests-e2e/core/gitopsdeployment_syncrun_test.go

```diff
@@ -470,6 +471,56 @@ var _ = Describe("GitOpsDeploymentSyncRun E2E tests", func() {
 				Expect(app.Status.History[i].Revision).Should(Equal(cr.revision))
 			}
 		})
+
+		It("should refresh before syncing the Application", func() {
+			By("add refresh annotation and verify if it'll be overwritten after refresh")
+			app := &appv1.Application{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      argocd.GenerateArgoCDApplicationName(string(gitOpsDeploymentResource.UID)),
+					Namespace: "gitops-service-argocd",
+				},
+			}
+			err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+				err := k8sClient.Get(ctx, client.ObjectKeyFromObject(app), app)
+				if err != nil {
+					return err
+				}
+				app.Annotations = make(map[string]string)
+				app.Annotations[appv1.AnnotationKeyRefresh] = string(appv1.RefreshTypeNormal)
+				return k8sClient.Update(ctx, app)
+			})
+			Expect(err).To(BeNil())
```

Comment:
Do we need this part of the test? I ask because I think creating the `GitOpsDeploymentSyncRun` below would also cause the `AnnotationKeyRefresh` to be added.

Reply:
I agree. It doesn't make sense and I have removed it from the test.